From b0ff8fb7a07603f22681a465a1143dc573f7a7dd Mon Sep 17 00:00:00 2001
From: pb01ka <gaganpb08singh@gmail.com>
Date: Thu, 21 May 2026 01:50:51 +0530
Subject: [PATCH] Compatible windows changes

---
 src/google/protobuf/message_lite.h | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/src/google/protobuf/message_lite.h b/src/google/protobuf/message_lite.h
index c416114aa..3994762f0 100644
--- a/src/google/protobuf/message_lite.h
+++ b/src/google/protobuf/message_lite.h
@@ -273,6 +273,12 @@ struct MessageTraitsImpl {
 template <typename T>
 using MessageTraits = decltype(MessageTraitsImpl::value<T>);
 
+#ifdef __CUDACC__
+struct EnumTraitsImpl {
+  template <typename T>
+  static std::enable_if_t<sizeof(T) != 0> value;
+};
+#else
 struct EnumTraitsImpl {
   struct Undefined;
   // We use an incomplete type to cause a compiler error if something tries to
@@ -282,6 +288,7 @@ struct EnumTraitsImpl {
   template <typename T>
   static std::enable_if_t<sizeof(T) != 0, Undefined> value;
 };
+#endif
 template <typename T>
 using EnumTraits = decltype(EnumTraitsImpl::value<T>);
 
